### DISCUSSION TITLE
Utilizing the regex pattern for custom fields fails, for every pattern I try.

### DISCUSSION THREAD

**Discussion by agoransson** [vendurehq/vendure#2853] [Q&A]
Hi

I've been tampering with custom fields lately and I find that the regular expression validation (pattern attribute) never works for me. I always get the following error:

`The custom field "issn" value ["1101-4369"] does not match the pattern [d{4}-d{3}[dxX]]`

I have tried with terminatoions as well (^ and $) but those fail as well. What am I missing?

```
    {
        name: 'issn',
        type: 'string',
        label: [ ... translations ... ],
        description: [ ... translations ... ],
        pattern: "\d{4}-\d{3}[\dxX]"
    },
```

I tried the value `1101-4369`

**Comment by michaelbromley** [comment]
Hi,

I believe this is because in the `\d` part of your regex, the `\` character is interpreted as an escape character, rather than a literal backslash. This is why the error message omits the backslash, making the regex look for the character `'d'` rather than a digit.

Try it like this: `pattern: "\\d{4}-\\d{3}[\\dxX]"`
